main/src/library/MutSet.flix

```diff
@@ -19,6 +19,11 @@
 ///
 opaque type MutSet[t] = Ref[Set[t]]
 
+instance Iterable[MutSet] {
+    pub def iterator(t: MutSet[a]): Iterator[a] & Impure =
+        t |> MutSet.toSet |> Set.toIterator
```

Comment:
Is this a problem too? To convert it to a set before making a iterator.

Reply:
I can see, that the `toIterator` function in `Set` converts the given `Set` to a list first.
```
    pub def toIterator(s: Set[a]): Iterator[a] & Impure =
        List.toIterator(Set.toList(s))
```

Reply:
@magnus-madsen

Reply:
I think that is fine, because behind the covers a mut set is just a set.

Reply:
So the conversion is free. (Check the definition of MutSet)

Reply:
ah, alright. That makes sense